### ISSUE TITLE
task.upload_artifact to wait until upload is finished before returning

### ISSUE THREAD

**Issue by H4dr1en** [allegroai/clearml#194]
Slack thread > https://allegroai-trains.slack.com/archives/CTK20V944/p1598346720094900

This is a proposal to extend `Task.upload_artifact` as follows:
```
def upload_artifact(..., wait_for_upload: bool = False):
    ...
    if wait_for_upload:
        self.flush(wait_for_uploads=True)
```

**Comment by bmartinn** [comment]
Thanks @H4dr1en !
I'll update here once we add the new functionality.
(My hope is to have better granularity, so you are not waiting for all uploads of the system, but specifically to the current artifact)

**Comment by allegroai-git** [comment]
Closing since related PR was merged and released ClearML/Trains v0.16.3.
Thanks for the PR! 😄
